commit da6567da7ea0e844c9a880e02c7b34e1a184edf8
Author: VAIBHAVPANT07 <vaibhavpant24@gmail.com>
Date:   Sat Apr 25 00:15:19 2026 +0530

    Fixed #37062 -- Added preserve_request support to RedirectView.

diff --git a/tests/generic_views/test_base.py b/tests/generic_views/test_base.py
index 9d6cef8a46..b1179ebe61 100644
--- a/tests/generic_views/test_base.py
+++ b/tests/generic_views/test_base.py
@@ -479,6 +479,20 @@ class RedirectViewTest(LoggingAssertionMixin, SimpleTestCase):
         self.assertEqual(response.status_code, 302)
         self.assertEqual(response.url, "/bar/")
 
+    def test_preserve_request_temporary_redirect(self):
+        response = RedirectView.as_view(url="/bar/", preserve_request=True)(
+            self.rf.get("/foo/")
+        )
+        self.assertEqual(response.status_code, 307)
+        self.assertEqual(response.url, "/bar/")
+
+    def test_preserve_request_permanent_redirect(self):
+        response = RedirectView.as_view(
+            url="/bar/", preserve_request=True, permanent=True
+        )(self.rf.get("/foo/"))
+        self.assertEqual(response.status_code, 308)
+        self.assertEqual(response.url, "/bar/")
+
     def test_include_args(self):
         "GET arguments can be included in the redirected URL"
         response = RedirectView.as_view(url="/bar/")(self.rf.get("/foo/"))
