commit 158fd81ef5a5647d27eb3065063284f9ee0a3ca4
Author: Amar <100243770+aadeina@users.noreply.github.com>
Date:   Tue Feb 3 00:34:04 2026 +0000

    Fixed #36899 -- Implemented SessionBase.__bool__.

diff --git a/django/contrib/sessions/backends/base.py b/django/contrib/sessions/backends/base.py
index e53f1d201a..8f673761f2 100644
--- a/django/contrib/sessions/backends/base.py
+++ b/django/contrib/sessions/backends/base.py
@@ -66,6 +66,9 @@ class SessionBase:
         del self._session[key]
         self.modified = True
 
+    def __bool__(self):
+        return not self.is_empty()
+
     @property
     def key_salt(self):
         return "django.contrib.sessions." + self.__class__.__qualname__
