commit e1df3950b2a3a852dfa9eb70738b7bac28cb0a0e
Author: Manas225 <manas112madeshiya@gmail.com>
Date:   Mon Mar 9 22:54:34 2026 +0530

    Fixed #36979 -- Made GenericInlineModelAdmin.get_formset() use get_exclude().

diff --git a/django/contrib/contenttypes/admin.py b/django/contrib/contenttypes/admin.py
index d324a4f4fe..c55325ecbe 100644
--- a/django/contrib/contenttypes/admin.py
+++ b/django/contrib/contenttypes/admin.py
@@ -100,12 +100,12 @@ class GenericInlineModelAdmin(InlineModelAdmin):
             fields = kwargs.pop("fields")
         else:
             fields = flatten_fieldsets(self.get_fieldsets(request, obj))
-        exclude = [*(self.exclude or []), *self.get_readonly_fields(request, obj)]
-        if (
-            self.exclude is None
-            and hasattr(self.form, "_meta")
-            and self.form._meta.exclude
-        ):
+        excluded = self.get_exclude(request, obj)
+        exclude = [
+            *(excluded or []),
+            *self.get_readonly_fields(request, obj),
+        ]
+        if excluded is None and hasattr(self.form, "_meta") and self.form._meta.exclude:
             # Take the custom ModelForm's Meta.exclude into account only if the
             # GenericInlineModelAdmin doesn't define its own.
             exclude.extend(self.form._meta.exclude)
