commit a486b48d4ea0d34babe58cae215b2123748aacbd
Author: vismaytiwari <vismay.t@gmail.com>
Date:   Thu Jul 9 11:15:01 2026 +0530

    Fixed #37210 -- Logged cached_db session cache delete failures.

diff --git a/tests/cache/failing_cache.py b/tests/cache/failing_cache.py
index 1c9b5996d6..089d732591 100644
--- a/tests/cache/failing_cache.py
+++ b/tests/cache/failing_cache.py
@@ -8,3 +8,9 @@ class CacheClass(LocMemCache):
 
     async def aset(self, *args, **kwargs):
         raise Exception("Faked exception saving to cache")
+
+    def delete(self, *args, **kwargs):
+        raise Exception("Faked exception deleting from cache")
+
+    async def adelete(self, *args, **kwargs):
+        raise Exception("Faked exception deleting from cache")
diff --git a/tests/sessions_tests/tests.py b/tests/sessions_tests/tests.py
index 1a4b114a8a..a7a07e5ea2 100644
--- a/tests/sessions_tests/tests.py
+++ b/tests/sessions_tests/tests.py
@@ -842,6 +842,23 @@ class CacheDBSessionTests(SessionTestsMixin, TestCase):
         self.assertEqual(log.message, f"Error saving to cache ({session._cache})")
         self.assertEqual(str(log.exc_info[1]), "Faked exception saving to cache")
 
+    @override_settings(
+        CACHES={"default": {"BACKEND": "cache.failing_cache.CacheClass"}}
+    )
+    def test_cache_delete_failure_non_fatal(self):
+        """Failing to delete from the cache does not raise errors."""
+        session = self.backend()
+        session.save()
+        session_key = session.session_key
+
+        with self.assertLogs("django.contrib.sessions", "ERROR") as cm:
+            session.delete(session_key)
+
+        # A proper ERROR log message was recorded.
+        log = cm.records[-1]
+        self.assertEqual(log.message, f"Error deleting from cache ({session._cache})")
+        self.assertEqual(str(log.exc_info[1]), "Faked exception deleting from cache")
+
     @override_settings(
         CACHES={"default": {"BACKEND": "cache.failing_cache.CacheClass"}}
     )
@@ -858,6 +875,23 @@ class CacheDBSessionTests(SessionTestsMixin, TestCase):
         self.assertEqual(log.message, f"Error saving to cache ({session._cache})")
         self.assertEqual(str(log.exc_info[1]), "Faked exception saving to cache")
 
+    @override_settings(
+        CACHES={"default": {"BACKEND": "cache.failing_cache.CacheClass"}}
+    )
+    async def test_cache_async_delete_failure_non_fatal(self):
+        """Failing to delete from the cache does not raise errors."""
+        session = self.backend()
+        await session.asave()
+        session_key = session.session_key
+
+        with self.assertLogs("django.contrib.sessions", "ERROR") as cm:
+            await session.adelete(session_key)
+
+        # A proper ERROR log message was recorded.
+        log = cm.records[-1]
+        self.assertEqual(log.message, f"Error deleting from cache ({session._cache})")
+        self.assertEqual(str(log.exc_info[1]), "Faked exception deleting from cache")
+
 
 @override_settings(USE_TZ=True)
 class CacheDBSessionWithTimeZoneTests(CacheDBSessionTests):
