commit ea9742c6d02edf64fc0969f21506f2048c976051
Author: Aaryan P <aaryanparik124@gmail.com>
Date:   Fri May 22 01:32:01 2026 +0530

    Fixed #37108 -- Made DjangoJSONEncoder consistently omit .000 microseconds.

diff --git a/django/core/serializers/json.py b/django/core/serializers/json.py
index b955939e0d..4b4f7f30c0 100644
--- a/django/core/serializers/json.py
+++ b/django/core/serializers/json.py
@@ -90,9 +90,10 @@ class DjangoJSONEncoder(json.JSONEncoder):
     def default(self, o):
         # See "Date Time String Format" in the ECMA-262 specification.
         if isinstance(o, datetime.datetime):
-            r = o.isoformat()
-            if o.microsecond:
-                r = r[:23] + r[26:]
+            r = o.isoformat(
+                sep="T",
+                timespec="milliseconds" if o.microsecond // 1000 else "seconds",
+            )
             if r.endswith("+00:00"):
                 r = r.removesuffix("+00:00") + "Z"
             return r
@@ -101,9 +102,9 @@ class DjangoJSONEncoder(json.JSONEncoder):
         elif isinstance(o, datetime.time):
             if is_aware(o):
                 raise ValueError("JSON can't represent timezone-aware times.")
-            r = o.isoformat()
-            if o.microsecond:
-                r = r[:12]
+            r = o.isoformat(
+                timespec="milliseconds" if o.microsecond // 1000 else "seconds"
+            )
             return r
         elif isinstance(o, datetime.timedelta):
             return duration_iso_string(o)
